client/tailscale/routes.go

```diff
@@ -93,3 +98,82 @@ func (c *Client) SetRoutes(ctx context.Context, deviceID string, subnets []netip
 	}
 	return srr, err
 }
+
+var (
+	ipv4default = netip.MustParsePrefix("0.0.0.0/0")
+	ipv6default = netip.MustParsePrefix("::/0")
+)
+
+func validateViaPrefix(ipp netip.Prefix) error {
+	if !tsaddr.IsViaPrefix(ipp) {
+		return fmt.Errorf("%v is not a 4-in-6 prefix", ipp)
+	}
+	if ipp.Bits() < (128 - 32) {
+		return fmt.Errorf("%v 4-in-6 prefix must be at least a /%v", ipp, 128-32)
+	}
+	a := ipp.Addr().As16()
+	// The first 64 bits of a are the via prefix.
+	// The next 32 bits are the "site ID".
+	// The last 32 bits are the IPv4.
+	// For now, we reserve the top 3 bytes of the site ID,
+	// and only allow users to use site IDs 0-255.
+	siteID := binary.BigEndian.Uint32(a[8:12])
+	if siteID > 0xFF {
+		return fmt.Errorf("route %v contains invalid site ID %08x; must be 0xff or less", ipp, siteID)
+	}
+	return nil
+}
+
+// CalcAdvertiseRoutes calculates the requested routes to be advertised by a node.
+// advertiseRoutes is the user-provided, comma-separated list of routes (IP addresses or CIDR prefixes) to advertise.
+// advertiseDefaultRoute indicates whether the node should act as an exit node and advertise default routes.
+func CalcAdvertiseRoutes(advertiseRoutes string, advertiseDefaultRoute bool) ([]netip.Prefix, error) {
```

Comment:
this is kinda bizarre public API for our client package which has a stronger API stability guarantee than the rest of the tree.

Can this live elsewhere?

Reply:
yeah, I didn't really *love* putting it in the client package.  I meant to ask in the PR message for suggestions of anywhere else that it might make sense to put it.  Any thoughts?

Reply:
okay, moved to `net/netutil`